### ISSUE TITLE
Support sequence with callback in YAML parser

### ISSUE THREAD

**Issue by alamar** [OpenHFT/Chronicle-Wire#609]
For the following yaml, TEXT parser allocates several gigabytes of off-heap and takes 5 seconds to parse; YAML parser fails to parse.

```
# Copyright 2021 - http://chronicle.software
!ChronicleServicesCfg {

  # defines all the chronicle queue and where they are stored
  queues: {

    # commands that are sent to the server from the web UI are all written to this queue
    web-gateway-out: { path: queues/replica/web-gateway-out, sourceId: 1 },

    # results of search queries, that are sent to the UI, are first written to this queue
    fix-search-out: { path: queues/replica/fix-search, sourceId: 2 },

    # for all the fix session, both the upstream and downstream messages are written to this queue
    fix-out: { path: queues/replica/fix-out, sourceId: 3 },

    #
    # PERIODIC UPDATES
    # ---------------

    # all period updates sent to the web gateway are written to this queue
    web-gateway-periodic-updates: { path: queues/replica/web-gateway-periodic-updates, sourceId: 4 },

    # all period updates sent to the fix engine are written to this queue
    fix-periodic-updates: { path: queues/replica/fix-periodic-updates, sourceId: 5 },

    #
    # ERROR REPORTING
    # ---------------

    # errors that may occur when fix session or routing rules are stored in the database
    fix-config-out: { path: queues/replica/fix-config-out, sourceId: 6 },

    # the errors that may occur when the ui changes the fix state is changes such as calling start / stop.
    session-state-updates: { path: queues/replica/session-state-updates, sourceId: 7 },

    # any errors that occur when writing the fix messages to the database are written to this queue
    db-out: { path: queues/replica/db-out, sourceId: 8 },

  },

  # defines all the chronicle services
  services: {

    # the webserver for orchestrating the fix sessions
    fix-web-gateway: {
      inputs: [ web-gateway-periodic-updates, session-state-updates, fix-config-out, fix-search-out ], # the response to events put on the command queue and response to
      output: web-gateway-out,  # the command queue to the fix-engine
      #hostname: 0.0.0.0, # must be either 0.0.0.0 or the address of the specific Docker container to run in the Docker environment
      #webport: 1080,
      #startFromStrategy: END, # upon a restart will only read new messages that have been written to the queue
      #prefix: "/", # this is used to prefix part to of the URL
      #webResources: "html/fixparser/build",
      implClass: !type software.chronicle.fix.gui.FixWebGatewayService,
#      fieldManifestClass: !type software.chronicle.tmp.fix50SP2.FieldManifest,      # use in the UI drop-down for auto complete of field names
      periodicUpdateMS: 1000, # used to drive updates to the websockets, checks if there is something in the queue once per millisecond
      # used in the UI drop downs when setting the fix version, to choose which MessageGenerator, MessageParserBase to MessageNotifier to use.

      fixToDto: !UnexpectedFieldHandlingFixDto {
        messageNotifierClass: !type software.chronicle.tmp.fix50SP2.messages.MessageNotifier,
      },
    }
  }
}
```

**Comment by alamar** [comment]
Memory usage is some issue affecting old JVMs and absent on Java 17. So, only YAML parser support is left.

**Comment by alamar** [comment]
```
{
  services: {
    fix-web-gateway: {
      # 534534

      inputs: #523523523
        [
            web-gateway-periodic-updates,

          # errwe
            session-state-updates,
            fix-config-out,
            fix-search-out
        ]
    }
  }
}

```

**Comment by hft-team-city** [comment]
Released in [Chronicle-Wire-2.24ea9](https://github.com/OpenHFT/Chronicle-Wire/releases/chronicle-wire-2.24ea9), [BOM-2.24ea41](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.24ea41)
